posts/2022-01-13-Rust-1.58.0.md

```diff
@@ -0,0 +1,184 @@
+   environment variable, if it was explicitly changed from the parent.
+2. The directory from which the application loaded.
+3. The current directory for the parent process.
+4. The 32-bit Windows system directory.
+5. The 16-bit Windows system directory.
+6. The Windows directory.
+7. The directories that are listed in the `PATH` environment variable.
+
+However, using the current directory can lead to surprising results, or even
+malicious behavior when dealing with untrusted directories. For example,
+`ripgrep` published [CVE-2021-3013] when they learned that their child
+processes could be intercepted in this way. Even Microsoft's own PowerShell
+[documents][PS] that they do not use the current directory for security.
+
+Rust now performs its own search without the current directory, and the legacy
+16-bit directory is also not included, as there is no API to discover its
+location. So the new `Command` search order for Rust on Windows is:
+
+1. The directories that are listed in the child's `PATH` environment variable.
+2. The directory from which the application loaded.
+3. The 32-bit Windows system directory.
+4. The Windows directory.
+5. The directories that are listed in the `PATH` environment variable.
+
+Non-Windows targets continue to use their platform-specific behavior, most
+often only considering the child or parent `PATH` environment variable.
+
+[`CreateProcess`]: https://docs.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-createprocessa
+[CVE-2021-3013]: https://www.cve.org/CVERecord?id=CVE-2021-3013
+[PS]: https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_command_precedence?view=powershell-7.2
+
+### More `#[must_use]` in the standard library
+
+The `#[must_use]` attribute can be applied to types or functions when failing
+to explicitly consider them or their output is almost certainly a bug. This has
+long been used in the standard library for types like `Result`, which should be
+checked for error conditions. This also helps catch mistakes such as expecting
+a function to mutate a value in-place, when it actually returns a new value.
+
+Library [proposal 35] was approved in October 2021 to audit and expand the
+application of `#[must_use]` throughout the standard library, covering many
+more functions where the primary effect is the return value. This is similar
+to the idea of function purity, but looser than a true language feature. Some
+of these additions were present in release 1.57.0, and now in 1.58.0 the effort
+has completed.
+
+[proposal 35]: https://github.com/rust-lang/libs-team/issues/35
+
+### Stabilized APIs
+
+The following methods and trait implementations were stabilized.
+
+- [`Metadata::is_symlink`]
+- [`Path::is_symlink`]
+- [`{integer}::saturating_div`]
+- [`Option::unwrap_unchecked`]
+- [`Result::unwrap_unchecked`]
+- [`Result::unwrap_err_unchecked`]
+- [`NonZero{unsigned}::is_power_of_two`]
```

Comment:
I think this landed in 1.59

https://github.com/rust-lang/rust/pull/91301

Reply:
Fixed.